scripts/dotnet-publish.sh

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+declare -a PARAM_BUILD_PROPS
+parsing_flag=""
+for arg in "$@"; do
+  if [[ "$arg" == "--env-vars" ]] || [[ "$arg" == "--build-props" ]]; then
+    parsing_flag="${arg:2}"
+  elif [[ "$parsing_flag" == "env-vars" ]]; then
+    export "${arg?}"
+  elif [[ "$parsing_flag" == "build-props" ]]; then
+    if [[ "$arg" == *\;* ]] && [[ "$arg" != *=\"*\" ]]; then
+      echo "Invalid BUILD_PROPS property: '""$arg""'." >&2
+      echo "To assign a list of values, the values must be enclosed with double quotes. For example: MyProperty=\"Value1;Value2\"." >&2
+      exit 1
+    fi
+    PARAM_BUILD_PROPS+=("$arg")
+  fi
```

Comment:
Should we warn about unrecognized arguments?

Reply:
These args are a way to pass the array parameters into the script. They are controlled by the Task itself. The `--env-vars`/`--build-props` act as placeholders between different parameters. We're ok to treat everything that is not a placeholder as a value.